addons/sale_stock/tests/test_anglosaxon_account.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+from odoo.addons.account.tests.account_test_savepoint import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+@tagged('post_install', '-at_install')
+class TestAngloSaxonAccounting(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company_data['company'].write({
+            'anglo_saxon_accounting': True
+        })
+        cls.company_data_2['company'].write({
+            'anglo_saxon_accounting': True
+        })
+        product_category = cls.env['product.category'].create({
+            'name': 'a random storable product category',
+            'property_cost_method': 'fifo',
+            'property_valuation': 'real_time',
+        })
+        cls.storable_product = cls.env['product.product'].create({
+            'name': 'storable product a',
+            'type': 'product',
+            'categ_id': product_category.id,
+        })
+        # Those values are company dependent and need to be explicitly set for both companies
+        cls.env.user.company_id = cls.company_data_2['company'].id
+        product_category.write({
+            'property_cost_method': 'fifo',
+            'property_valuation': 'real_time',
+        })
+        cls.env.user.company_id = cls.company_data['company'].id
```

Comment:
Don't do that. `product_category.with_company(force_company=cls.company_data_2['company'].id).write({...})` instead